impeller/typographer/backends/stb/typographer_context_stb.cc

```diff
@@ -419,24 +469,7 @@ std::shared_ptr<GlyphAtlas> TypographerContextSTB::CreateGlyphAtlas(
   // ---------------------------------------------------------------------------
   std::vector<FontGlyphPair> new_glyphs;
   std::vector<Rect> new_sizes;
-  for (const auto& font_value : font_glyph_map) {
-    const ScaledFont& scaled_font = font_value.first;
-    const FontGlyphAtlas* font_glyph_atlas =
-        last_atlas->GetFontGlyphAtlas(scaled_font.font, scaled_font.scale);
-    if (font_glyph_atlas) {
-      for (const SubpixelGlyph& glyph : font_value.second) {
-        if (!font_glyph_atlas->FindGlyphBounds(glyph)) {
-          new_glyphs.emplace_back(scaled_font, glyph);
-          new_sizes.push_back(ComputeGlyphSize(scaled_font, glyph));
-        }
-      }
-    } else {
-      for (const SubpixelGlyph& glyph : font_value.second) {
-        new_glyphs.emplace_back(scaled_font, glyph);
-        new_sizes.push_back(ComputeGlyphSize(scaled_font, glyph));
-      }
-    }
-  }
+  CollectNewGlyphs(last_atlas, text_frames, new_glyphs, new_sizes);
```

Comment:
DCHECK that the sizes of the two out vectors are the same?

Reply:
I'm slightly partial to returning a pair of vectors instead of out variables because it makes it clear that existing values in the vectors will be not be disregarded. I didn't see a place where you didn't have to clear the vectors before calling CollectNewGlyphs. Just one less thing you don't have to remember to do. Then you can use a structured binding to do something like:

```c++
auto [new_glyphs, new_sizes] = CollectNewGlyphs(last_atlas, text_frames);
```

Reply:
Done. I think I one point we were incrementally adding to these but that is long since gone